### PR TITLE
core: add nil checks

### PR DESCRIPTION
**Pull request by dB2510**
Adds nil checks before reading proto messages in core package mentioned in audit report.

Refer: OBOL-03

category: misc
ticket: #1997

### CODE REVIEW

File: core/proto.go [hunk 1]
```diff
@@ -179,6 +187,10 @@ func UnsignedDataSetToProto(set UnsignedDataSet) (*pbv1.UnsignedDataSet, error)
 
 // UnsignedDataSetFromProto returns the set from a protobuf.
 func UnsignedDataSetFromProto(typ DutyType, set *pbv1.UnsignedDataSet) (UnsignedDataSet, error) {
+	if set == nil {
+		return nil, errors.New("unsigned data set proto cannot be nil")
+	}
+
 	resp := make(UnsignedDataSet)
 	for pubkey, data := range set.Set {
```
Comment (corverroos) [inline core/proto.go:195]: what about set.Set?

**Review by corverroos — APPROVED** [review]

File: core/proto.go (hunk 1 above)
Comment (dB2510) [reply core/proto.go:195]: for range loop over set.Set will take care of that, so no panic. However, I can add explicit check for that too.

Comment (dB2510) [reply core/proto.go:196]: Done!
Also added some tests.

**Comment by dB2510** [comment]
@corverroos 
A short note: The audit report also mentioned to add check in `core/proto.go::DutyFromProto()`. I didn't added that as we already do nil checks before passing duty proto to this function wherever it is called. Also adding check to this will modify function signature to return error which impacts uses of this function some other places as well. Not worth it IMO. WDYT?

File: core/proto.go (hunk 1 above)
Comment (corverroos) [reply core/proto.go:196]: as, set is a map, in that case, probably better to check for len(set.Set) == 0

**Pull request merged**